sqlmesh/dbt/adapter.py

```diff
@@ -297,44 +308,50 @@ def execute(
         return AdapterResponse("Success"), empty_table()
 
     def resolve_schema(self, relation: BaseRelation) -> t.Optional[str]:
-        schema = self._map_table_name(relation.database, relation.schema, relation.identifier).db
-        if not schema:
-            return None
-        return schema
+        schema = self._map_table_name(self._normalize(self._relation_to_table(relation))).db
+        return schema if schema else None
 
     def resolve_identifier(self, relation: BaseRelation) -> t.Optional[str]:
-        identifier = self._map_table_name(
-            relation.database, relation.schema, relation.identifier
-        ).name
-        if not identifier:
-            return None
-        return identifier
-
-    def _map_table_name(
-        self, database: t.Optional[str], schema: t.Optional[str], identifier: t.Optional[str]
-    ) -> exp.Table:
-        name = normalize_model_name(
-            exp.table_(identifier or "", db=schema, catalog=database),
-            dialect=self.engine_adapter.dialect,
-        )
+        identifier = self._map_table_name(self._normalize(self._relation_to_table(relation))).name
+        return identifier if identifier else None
+
+    def _map_table_name(self, table: exp.Table) -> exp.Table:
+        name = table.sql(dialect=self.engine_adapter.dialect)
```

Comment:
is the table_mapping normalized model names? if so this shouldn't have dialect

Reply:
Yep nice catch